src/metrics/index.ts

```diff
@@ -0,0 +1,104 @@
+import { MeticsObjectType, SharedMeticsObjectType } from '../utils/common';
+import sdkDetails from '../../package.json';
+import { getMetaObject } from '../utils/helpers';
+
+export const METRIC_OBJECT: SharedMeticsObjectType = { records: [] };
+
+export function setBearerToken(metadata: any) {
+  metadata?.clientJSON?.config?.getBearerToken?.().then((authToken: string) => {
+    METRIC_OBJECT.bearerToken = authToken;
+  }).catch((err: any) => {
+    // eslint-disable-next-line no-console
+    console.error(err);
+  });
+}
+
+export function initalizeMetricObject(metadata: any, elementId: string) {
+  setBearerToken(metadata);
```

Comment:
Don't think this is the right approach for setting bearer token.
There will be cases where this approach fail.
1. what if the bearer token expires before sending the event request ?
2. what if the event request is emitted before bearer token is set here ?

Reply:
1.If the bearer token expires this breaks then our collect call and all other will also fail. 
2.We are not emitting the request if there is[ no bearer token](https://github.com/skyflowapi/skyflow-js/blob/847a4d85f1975ad5b0506b860ba535fc5f4967b3/src/metrics/index.ts#L82).